expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java

```diff
@@ -222,19 +223,31 @@ void shouldReturnFramedIndexDocumentFromResource(Class<?> instanceType)
     void shouldReturnIndexDocumentContainingLicense(String licenseUri, LicenseType expectedLicense)
         throws JsonProcessingException, NotFoundException {
         var fileWithLicense = randomPublishedFile(licenseUri);
+        var associatedLink = new AssociatedLink(randomUri(), null, null);
         var publication = PublicationGenerator.randomPublication()
                                       .copy()
-                                      .withAssociatedArtifacts(List.of(fileWithLicense))
+                                      .withAssociatedArtifacts(List.of(fileWithLicense, associatedLink))
                                       .build();
 
         var resourceUpdate = Resource.fromPublication(publication);
         var indexDoc = (ExpandedResource) expansionService.expandEntry(resourceUpdate);
-        var licensesAsString = indexDoc.asJsonNode().get("associatedArtifacts").get(0).get("license").toString();
+        var licensesAsString = getLicenseForFile(indexDoc);
         var license = JsonUtils.dtoObjectMapper.readValue(licensesAsString, License.class);
 
         assertThat(license.name(), is(equalTo(expectedLicense.toLicense(URI.create(licenseUri)).name())));
     }
 
+    private static String getLicenseForFile(ExpandedResource indexDoc) {
+        var associatedArtifacts = indexDoc.asJsonNode().get("associatedArtifacts");
+        String string = null;
+        for (JsonNode artifact : associatedArtifacts) {
+            if (!artifact.get("type").asText().equals("AssociatedLink")) {
+                string = artifact.get("license").toString();
+            }
```

Comment:
It might be a good idea to check for Null both places where you look up nodes in this block?

Reply:
This is a test code, but I have fixed it :)